### DISCUSSION TITLE
The NetBeans Ant Build Failed with gradle (JDK 19)

### DISCUSSION THREAD

**Discussion by WhiteHsu** [apache/netbeans#4801] [Q&A] [answered]
Hello Guys,

I'm trying to build the project but encounter another failure with gradle. May I consult your opinions?

What I've done are:

1. Fork and clone the project to my local on my Windows 10.
2. Install JDK 19 (The Latest Version)
3. Download ANT 1.10.12 (Latest Version too)
4. Then go to the local dir of the home folder netbeans
5. Run command "ant" with Admin auth.

However after running the build command "ant" with 15 min, it got build failure while running Gradle. 
![image](https://user-images.githubusercontent.com/7831418/196045526-dffcc21f-cb97-4eda-b67c-e35394299dc7.png)
![image](https://user-images.githubusercontent.com/7831418/196045540-d76567d8-3210-4567-ac96-04c89f47bc5c.png)

I've tried to install Gradle 7.5.1 and rerun the ant, but still gotten the same build failure.

May you kindly suggest what I may miss? Thank you in advance.

**Comment by mbien** [comment]
Hi @WhiteHsu,

I don't have a windows setup ready, but I usually build with
`ant -Dcluster.config=full`
That is also how the CI builds it. (However CI only builds on 11 and 17).

my build script looks something like:
```bash
export JAVA_HOME=/home/mbien/dev/java/jdk-19.0.0.36.1-corretto
export PATH=$JAVA_HOME/bin:$PATH

echo nbjdk.home=$JAVA_HOME > nbbuild/user.build.properties
java -version
ant -version

ant clean -q
ant -q build-nozip -Dcluster.config=full
```

I just built with JDK 19 and it worked
```
BUILD SUCCESSFUL
Total time: 7 minutes 47 seconds
```

Maybe you could try again with supplying the cluster config (if it still fails try on JDK 17).

edit: see https://github.com/apache/netbeans/discussions/4801#discussioncomment-4247604 and https://github.com/apache/netbeans/discussions/4801#discussioncomment-4247869

**Comment by lkishalmi** [comment, marked as answer]
The Gradle Tooling is built with Gradle 7.4, that does not support Java 19.

Build the IDE with Java 11 or Java 17. Those should work.

**Reply by mbien** [reply]
there is also the option to tell gradle to run the daemon on a gradle compatible JDK while building with JDK 19. That is how CI is set up atm.

add `org.gradle.java.home=` <path to JDK 11> to `~/.gradle/gradle.properties`

I don't know where the file is on windows but gradle doc will probably have that information

Gradle is caching a few things. I noticed that I had to do the same locally once I cleared `.gradle`. But before clearing it, NetBeans built just fine on JDK 19 for me - that is why I couldn't reproduce the issue. I am still not 100% sure what is going on there exactly since I never set `org.gradle.java.home` before.
